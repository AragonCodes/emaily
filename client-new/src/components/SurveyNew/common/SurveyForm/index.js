import SurveyFormComponent from './SurveyForm';

export default SurveyFormComponent;