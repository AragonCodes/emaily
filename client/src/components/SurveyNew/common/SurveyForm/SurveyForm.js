import React from 'react';
import { reduxForm, Field } from 'redux-form';
import { Link } from 'react-router-dom';
import SurveyField from './SurveyField';
import validateEmails from '../../../../utils/validateEmails';
import formFields from '../formFields';

function SurveyForm({ handleSubmit, onSurveySubmit }) {
	function renderFields() {
		return formFields.map(({ label, name }) => (
			<Field
				key={name}
				component={SurveyField}
				type="text"
				label={label}
				name={name}
			/>
		));
	}

	return (
		<div>
			<form onSubmit={handleSubmit(onSurveySubmit)}>
				{renderFields()}
				<Link to="/surveys" className="grey lighten-1 btn-flat white-text">
					Cancel
				</Link>
				<button type="submit" className="red accent-2 btn-flat right white-text">
					Next
					<i className="material-icons right">done</i>
				</button>
			</form>
		</div>
	);
}

function validate(values) {
	const errors = {};

	errors.recipients = validateEmails(values.recipients || '');

	formFields.forEach(({ name }) => {
		if (!values[name])
			errors[name] = 'You must provide a value';
	});

	return errors;
}

export default reduxForm({
	validate,
	form: 'surveyForm',
	destroyOnUnmount: false
})(SurveyForm);
