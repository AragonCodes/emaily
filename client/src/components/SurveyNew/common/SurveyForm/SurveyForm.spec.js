import React from 'react';
import { shallow } from 'enzyme';
import SurveyForm from './SurveyForm';

describe('SurveyForm', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<SurveyForm
				handleSubmit={jest.fn()}
				onSurveySubmit={jest.fn()}
			/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});