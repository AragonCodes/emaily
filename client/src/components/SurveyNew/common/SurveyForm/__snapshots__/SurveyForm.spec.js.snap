// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyForm should render correctly 1`] = `"<Hoc handleSubmit={[Function: mockConstructor]} onSurveySubmit={[Function: mockConstructor]} touchOnBlur={true} touchOnChange={false} persistentSubmitErrors={false} destroyOnUnmount={false} shouldAsyncValidate={[Function: defaultShouldAsyncValidate]} shouldValidate={[Function: defaultShouldValidate]} shouldError={[Function: defaultShouldError]} shouldWarn={[Function: defaultShouldWarn]} enableReinitialize={false} keepDirtyOnReinitialize={false} updateUnregisteredFields={false} getFormState={[Function: getFormState]} pure={true} forceUnregisterOnUnmount={false} submitAsSideEffect={false} validate={[Function: validate]} form=\\"surveyForm\\" forwardedRef={{...}} />"`;
